input/input.go

```diff
@@ -29,10 +31,21 @@ type DefaultHandler struct {
 
 	metrics     mdata.Metrics
 	metricIndex idx.MetricIndex
+
+	buffer sync.Map
+}
+
+type IntervalLookupRecord struct {
+	sync.Mutex
+
+	Interval   int
+	DataPoints []*schema.MetricData
+	Last       time.Time
+	pos        int
```

Comment:
the values going into the map should be as small & pointerless as possible, for two reasons: memory usage and GC workload.
we don't need the full metricdata of each datapoint, we don't need a lock for each individual record (the logic is so concise that a lock on just the map is enough), pos is trivially implied (check whether the values in the array are 0), and time.Time is a fairly bloated structure (in particular, holds a pointer to a timezone)

thus, instead of a sync.Map holding these items, I suggest one of these two:
```
map[string][4]uint32
```
```
type record struct {
interval uint32
points [4]uint32
}

map[string]record
```